interp/interp.go

```diff
@@ -335,9 +336,29 @@ func (interp *Interpreter) Repl(in, out *os.File) {
 	}
 }
 
+// wrapper around Eval that traps a panic if the input is a terminal
+func (interp *Interpreter) TrapEval(in *os.File, src string) (reflect.Value, error) {
+	if runningInTerminal(in) {
```

Comment:
Can be simplify like that:
```go
if in == os.Stdin {
//...
}
```

Reply:
Alright, I'll un-export. I did not see your reply here. I was using the same check that was already present. (for the prompt). I just combined the check into on one function....

Reply:
yes it's not directly related to your changes, it's something that I have missing previously.

Reply:
you can keep like that but you can simplify the function:
```go
func runningInTerminal(in *os.File) bool {
	stat, err := in.Stat()
	return err == nil && stat.Mode()&os.ModeCharDevice != 0
}
```

Reply:
I just tried it with
```go
if in == os.Stdin {
//...
}
```
That causes yaegi to keep running after a panic when stdin is redirected from a file.

Reply:
Alright, so two PRs now, one with TrapEval and one with trapEval and both are passing.

Reply:
Maybe the check should only be done once. (for both the prompt and the eval)
Unless there is some common condition where stdin starts out as Terminal input, but then changes to non Terminal input.

Reply:
ok good catch.

could you just simplify the function?

```go
func runningInTerminal(in *os.File) bool {
	stat, err := in.Stat()
	return err == nil && stat.Mode()&os.ModeCharDevice != 0
}
``

Reply:
Oh, the prompt check was only once. I'm making the trapEval check it only once as well.